src/javascript/app_2/pages/trading/components/purchase.jsx

```diff
@@ -1,28 +1,47 @@
-import React           from 'react';
-import PropTypes       from 'prop-types';
-import { toGMTFormat } from '../../../common/date_time';
-import Button          from '../../../components/form/button.jsx';
-import Fieldset        from '../../../components/form/fieldset.jsx';
-import { connect }     from '../../../store/connect';
-import { isLoggedIn }  from '../../../../_common/base/client_base';
-import { localize }    from '../../../../_common/localize';
+import React               from 'react';
+import PropTypes           from 'prop-types';
+import { toGMTFormat }     from '../../../common/date_time';
+import Button              from '../../../components/form/button.jsx';
+import Tooltip             from '../../../components/elements/tooltip.jsx';
+import UILoader            from '../../../components/elements/ui_loader.jsx';
+import Fieldset            from '../../../components/form/fieldset.jsx';
+import { connect }         from '../../../store/connect';
+import Client,
+       { isLoggedIn }      from '../../../../_common/base/client_base';
+import { redirectToLogin } from '../../../../_common/base/login';
+import { localize }        from '../../../../_common/localize';
+import Url                 from '../../../../_common/url';
 import {
     getPropertyValue,
-    isEmptyObject }    from '../../../../_common/utility';
+    isEmptyObject }        from '../../../../_common/utility';
 
 // TODO: update/remove this once the design is ready
 const createPurchaseInfo = (purchase_info) => (
     <React.Fragment>
-        <div><strong>Purchase Info:</strong></div>
-        <div>Buy Price: {purchase_info.buy_price}</div>
-        <div>Payout: {purchase_info.payout}</div>
-        <div>Start: {toGMTFormat(purchase_info.start_time * 1000)}</div>
-        <div>Contract ID: {purchase_info.contract_id}</div>
-        <div>Transaction ID: {purchase_info.transaction_id}</div>
-        <div>Description: {purchase_info.longcode}</div>
+        <div><strong>{localize('Purchase Info')}:</strong></div>
+        <div>{localize('Buy Price')}: {purchase_info.buy_price}</div>
+        <div>{localize('Payout')}: {purchase_info.payout}</div>
+        <div>{localize('Start')}: {toGMTFormat(purchase_info.start_time * 1000)}</div>
+        <div>{localize('Contract ID')}: {purchase_info.contract_id}</div>
+        <div>{localize('Transaction ID')}: {purchase_info.transaction_id}</div>
+        <div>{localize('Description')}: {purchase_info.longcode}</div>
     </React.Fragment>
 );
 
+const LogInMessage = () => (
+    <div className='purchase-login-wrapper'>
+        <span className='info-text'>{localize('Please login to purchase the contract  ')}</span>
+        <Button
+            className='secondary orange'
+            has_effect
+            text={localize('LOGIN')}
```

Comment:
same here, I think it should be `log in`. can we make the text into uppercase using CSS instead, to avoid duplication for translators?